File: 02.CLIESC/ConUni_Restfull_Dotnet_ClienteEscritorio_Gr06/src/ec/edu/monster/controlador/ConversionControlador.java
package ec.edu.monster.controlador;

import ec.edu.monster.modelo.ConversionModelo;

public class ConversionControlador {

    private final ConversionModelo modelo;

    public ConversionControlador() {
        this.modelo = new ConversionModelo();
    }

    public boolean validarLogin(String usuario, String clave) {
        try {
            return modelo.login(usuario, clave);
        } catch (Exception e) {
            System.err.println("Error al validar login: " + e.getMessage());
            return false;
        }
    }

    public double convertirCmAPulgadas(double cm) {
        try {
            return modelo.centimetrosAPulgadas(cm);
        } catch (Exception e) {
            System.err.println("Error en la conversión: " + e.getMessage());
            return -1;
        }
    }

    public double convertirPulgadasACm(double pulgadas) {
        try {
            return modelo.pulgadasACentimetros(pulgadas);
        } catch (Exception e) {
            System.err.println("Error en la conversión: " + e.getMessage());
            return -1;
        }
    }
}

File: 02. CLIWEB/WS_ConversionUnidades_RESTFUL_ClienteWeb/src/ec.edu.monster.controlador/ConversionControlador.js
import { login, convertirPulgadasACm, convertirCmAPulgadas } from "../ec.edu.monster.modelo/ConversionServicio";

export const manejarLogin = async (usuario, clave, setAutenticado, setError) => {
  try {
    const isValid = await login(usuario, clave);
    if (isValid) {
      setAutenticado(true);
    } else {
      throw new Error("Credenciales incorrectas. Intente nuevamente.");
    }
  } catch (error) {
    setError(error.message);
  }
};

export const manejarConversion = async (tipo, valor, setResultado, setError) => {
  try {
    let resultado;
    if (tipo === "pulgadasACm") {
      resultado = await convertirPulgadasACm(valor);
    } else if (tipo === "cmAPulgadas") {
      resultado = await convertirCmAPulgadas(valor);
    }
    setResultado(resultado);
  } catch (error) {
    setError(error.message);
  }
};

